### PR TITLE
Feature/stale cache wrapper

### PR DESCRIPTION
**Pull request by toka03**
Create StaleableCache that wraps other cache object and provides functionality for store and serve staled values in cache

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -242,6 +242,72 @@ cacheManager := cache.NewLoadable[*Book](
...
+but then performing a refresh in the background to ensure cached data is always available to the caller.
+
+```go
+// Initialise any other cache
+c := cache.New[string]()
+
+// Wrap the new cache with the StaleableCache
+ws := cache.NewStaleable[string](
+	c, 
+	cache.WithTTL[string](10*time.Minute),
```
Comment (azarc-jono-langley) [inline README.md:258]: Is this default ttl?

File: cache/cache_test.go [hunk 2]
```diff
@@ -206,36 +206,22 @@ func TestCacheGetType(t *testing.T) {
...
 	// When
-	computedKey := cache.getCacheKey("my-Key")
+	computedKey := getCacheKey("my-Key")
 
 	// Then
 	assert.Equal(t, "my-Key", computedKey)
 }
 
 func TestCacheGetCacheKeyWhenKeyIsStruct(t *testing.T) {
-	// Given
```
Comment (azarc-jono-langley) [inline cache/cache_test.go:224]: sticking to the original convention, this should have a given?

File: cache/cache_test.go [hunk 3]
```diff
@@ -248,17 +234,10 @@ func (_ *StructWithGenerator) GetCacheKey() string {
 }
 
 func TestCacheGetCacheKeyWhenKeyImplementsGenerator(t *testing.T) {
-	// Given
```
Comment (azarc-jono-langley) [inline cache/cache_test.go:251]: same as above.

**Review by azarc-jono-langley — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (toka03) [reply README.md:258]: WithTTL provides ttl for how much time a record should stay active. If you don't provide it, value will be 0 and the underlying store ttl will be used by default

File: cache/cache_test.go (hunk 2 above)
Comment (toka03) [reply cache/cache_test.go:224]: nothing is given, it just tests a standalone `getCacheKey` function. There is no preconfiguration needed

File: cache/cache_test.go (hunk 3 above)
Comment (toka03) [reply cache/cache_test.go:251]: same as above :)

**Pull request merged**